tests/manager/data_layer/test_bibliographic.py

```diff
@@ -1163,3 +1163,64 @@ def test_data_source_last_updated_updates_timestamp(
         with freeze_time(now):
             bibliographic.apply(db.session, edition, None)
         assert edition.updated_at == now
+
+    def test_validate_primary_identifier_case_insensitive(
+        self, db: DatabaseTransactionFixture
+    ):
+        """Test that _validate_primary_identifier performs case-insensitive comparison."""
+        # Create an edition with a mixed-case identifier
+        edition = db.edition()
+        edition.primary_identifier.identifier = "ABC123def"
+        db.session.flush()
+
+        identifier_lower = edition.primary_identifier.identifier.lower()
+        identifier_upper = edition.primary_identifier.identifier.upper()
+        identifier_mixed = edition.primary_identifier.identifier.swapcase()
+
+        # Test with lowercase version - should NOT raise an error
```

Comment:
Minor: Maybe say why it should not cause an error (because the identifiers match exactly -- no difference in case)